src/Telemetry/AzureDevOpsTimeline/AzureDevOpsTimeline.cs

```diff
@@ -332,6 +337,33 @@ private async Task RunLoop(CancellationToken cancellationToken)
                     new KustoValue("Message", b.Raw.Message, KustoDataTypes.String),
                     new KustoValue("Bucket", b.Bucket, KustoDataTypes.String),
                 });
+
+            List<BuildPath> longestBuildPaths = new List<BuildPath>();
+
+            foreach (var build in builds)
+            {
+                MaestroBuild root = _context.Builds.Where(b => b.AzureDevOpsBuildId == build.Id).FirstOrDefault();
+                if (root != null)
+                {
+                    longestBuildPaths.Add(await GetLongestBuildPath(root, new List<string>(), builds, options));
+                }
+            }
+
+            BuildPath[] buildPaths = longestBuildPaths.ToArray();
```

Comment:
Why... to array? This helper just expects an IEnumerable... the list itself would have been sufficient.